File: src/engine/commands/killcommand.cpp
#include "killcommand.h"

#include "util.h"


#define super Command

KillCommand::KillCommand(QObject *parent) :
    super(parent) {

    setDescription("Attacks another character.\n"
                   "\n"
                   "Examples: kill joe, attack earl");
}

KillCommand::~KillCommand() {
}

void KillCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    GameObjectPtr character = takeObject(currentRoom()->characters());
    if (!requireSome(character, "Kill who?")) {
        return;
    }

    player->kill(character);
}

File: data/commands/killcommand.js
function KillCommand() {
    Command.call(this);

    this.setDescription("Attack another character.\n" +
                        "\n" +
                        "Examples: kill joe, attack earl");
}

KillCommand.prototype = new Command();
KillCommand.prototype.constructor = KillCommand;

KillCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var character = this.takeObject(this.currentRoom.characters);
    if (!this.requireSome(character, "Kill who?")) {
        return;
    }

    player.kill(character);
};

CommandRegistry.registerCommand("attack", new KillCommand());
CommandRegistry.registerCommand("kill", new KillCommand());
